src/page-modules/assistant/layout.tsx

```diff
@@ -107,9 +108,11 @@ function AssistantLayout({ children, tripQuery }: AssistantLayoutProps) {
   const onToSelected = async (to: GeocoderFeature) =>
     setValuesWithLoading({ to });
   const onTransportFilterChanged = debounce(
-    async (transportModeFilter: string[] | null) =>
-      setValuesWithLoading({ transportModeFilter }, true),
-    500,
+    async (transportModeFilter: string[] | null) => {
+      setNoneFilters(transportModeFilter?.length === 0 ? true : false);
```

Comment:
Can't this be a derived state, from `transportModeFilter`, or is it necessary to have its own state?

Reply:
I belive it is necessary to have its own state. This was the only way is was able to uncheck all travel mode filters and update the URL accordingly😅

Reply:
Isn't possible to just check the length of `transportModeFilter` where `!noneFilters` is checked for? 

I get that the `noneFilters` is used to trigger the `useEffect`, but wouldn't it be sufficient to either use the `transportModeFilter` in the condition and set it to `null` in the else clause? Or just pass `null` in `onTransportFilterChanged` if `transportModeFilter` is not present?